File: todo-service/src/main/java/se/fredf/app/todo/TodoController.java
//package se.fredf.app.todo;
//
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.http.MediaType;
//import org.springframework.web.bind.annotation.*;
//import reactor.core.publisher.Mono;
//
//@RestController
//@RequestMapping("/todos")
//public class TodoController {
//
//    private final TodoService todoService;
//
//    @Autowired
//    public TodoController(TodoService todoService) {
//        this.todoService = todoService;
//    }
//
//    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE)
//    Todo create(@RequestBody final Todo req) {
//        return todoService.create(req);
//    }
//
//    @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
//    Mono<Todo> getById(@PathVariable("id") final long id) {
//        return todoService.rxGetById(id);
//    }
//
//    @PatchMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
//    Todo updateTodo(@PathVariable("id") final long id, @RequestBody final UpdateTodoRequest req) {
//        return todoService.updateTodo(id, req);
//    }
//
//}

File: todo-service/src/main/kotlin/se/fredf/app/todo/TodoController.kt
package se.fredf.app.todo

import io.swagger.annotations.ApiOperation
import io.swagger.annotations.ApiResponse
import io.swagger.annotations.ApiResponses
import org.springframework.http.HttpStatus
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/todos")
class TodoController(private val todoService: TodoService) {

    @PostMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
    @ResponseStatus(HttpStatus.CREATED)
    fun create(@RequestBody req: CreateTodoRequest): Todo = todoService.create(req)

    @GetMapping("/{id}", produces = [MediaType.APPLICATION_JSON_VALUE])
    @ResponseStatus(HttpStatus.OK)
    fun getById(@PathVariable("id") id: Long): Todo = todoService.getByIdKtl(id)

    @PatchMapping("/{id}", produces = [MediaType.APPLICATION_JSON_VALUE])
    @ResponseStatus(HttpStatus.OK)
    fun updateTodo(@PathVariable("id") id: Long, @RequestBody req: UpdateTodoRequest): Todo =
        todoService.updateTodo(id, req)


}